addon/route-handler.js

```diff
@@ -25,10 +24,8 @@ function createHandler({ verb, schema, serializerOrRegistry, rawHandler, options
   let handler;
   let args = [ schema, serializerOrRegistry, rawHandler, options ];
   let type = typeOf(rawHandler);
-  if (type === 'function') {
-    handler = new FunctionHandler(...args);
-  } else if (type === 'object') {
-    handler = new ObjectHandler(...args);
+  if (type === 'function' || type === 'object') {
+    handler = new BaseHandler(...args);
```

Comment:
I'm wary of using this base class here because, it's a *shorthand* base route handler, but the function/object route handlers aren't shorthands. Shorthand means something specific, it means you either passed a string, array, or nothing (undefined) after the path parameter. So I'm wary of using it for function/object because there's a lot of logic there that doesn't apply.